utils/lib/rr_graph/graph.py

```diff
@@ -1680,12 +1806,15 @@ def edges_for_node(self, xml_node):
         list of RoutingEdge
             XML edges (ET._Element) associated with the given node.
 
-        Example
-        -------
+        Examples
+        --------
+        >>> r = simple_test_routing()
+        >>> [RoutingGraphPrinter.edge(r, e) for e in r.edges_for_node(r.get_node_by_id(1))]
+        ['X000Y000[00].SRC--> ->>- X000Y000[00].R-PIN>', 'X000Y000[00].R-PIN> ->>- X000Y000<-00->X000Y010']
+        >>> [RoutingGraphPrinter.edge(r, e) for e in r.edges_for_node(r.get_node_by_id(2))]
+        ['X000Y000[00].R-PIN> ->>- X000Y000<-00->X000Y010', 'X000Y000<-00->X000Y010 ->>- X000Y010[00].L-PIN<']
         """
-        node_id = xml_node.attrib.get('id', None)
-        assert node_id is not None, ET.tostring(xml_node)
-        return self.edges_for_allnodes()[node_id]
+        return self.edges_for_allnodes()[self._get_xml_id(xml_node)]
```

Comment:
Is this quick? If not, might be good to give a warning in the function header

Reply:
Done!